From 6912692a939a87db07e779ce2814c180693e758b Mon Sep 17 00:00:00 2001
From: svenate <salini.venate@intel.com>
Date: Wed, 30 Mar 2022 13:05:23 +0530
Subject: [PATCH] Fix fscanf overflow error

Fix the below fscanf overflow error:
error: 'fscanf' may overflow; destination buffer in argument 4
has size 24, but the corresponding specifier may require size
25 [-Werror,-Wfortify-source]
  fscanf(to, "#V1.0 %16s%24s\n", name, lastbootuptime);

Signed-off-by: svenate <salini.venate@intel.com>
---
 crashlog/history.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/crashlog/history.c b/crashlog/history.c
index 5e8ab24..273434c 100644
--- a/crashlog/history.c
+++ b/crashlog/history.c
@@ -49,7 +49,7 @@ static int nextline = -1;
 static unsigned int fileentries = 0;
 static int loop_uptime_event = 1;
 /* last uptime value set at device boot only */
-static char lastbootuptime[24] = "0000:00:00";
+static char lastbootuptime[25] = "0000:00:00";
 static char lastfakeprop[PROPERTY_VALUE_MAX] = "";
 extern int gabortcleansd;
 // global variable to enable dynamic change of uptime frequency
-- 
2.17.1

